tee-worker/litentry/core/assertion-build/src/a10.rs

```diff
@@ -43,60 +44,33 @@ pub fn build(req: &AssertionBuildRequest, min_balance: ParameterString) -> Resul
 
 	let mut client = AchainableClient::new();
 	let identities = transpose_identity(&req.identities);
+	let addresses = identities
+		.into_iter()
+		.filter(|(network, _)| *network == Web3Network::Ethereum)
```

Comment:
I don't think you need the `filter` here => it should have the desired network types already, see `get_supported_web3networks` and its usage.

Same applies to other assertion building.

Reply:
Here is to filter the address set of the specified network, because different assertions support different networks.

Reply:
Yes I mean:
> it should have the desired network types already, see get_supported_web3networks and its usage.

Did you check this:

https://github.com/litentry/litentry-parachain/blob/61f77ac1f57e0197b214b17231b5dcf076c3910b/primitives/core/src/assertion.rs#L59-L75

and this:

https://github.com/litentry/litentry-parachain/blob/61f77ac1f57e0197b214b17231b5dcf076c3910b/tee-worker/app-libs/stf/src/trusted_call_litentry.rs#L148

... so it's redundant unless we are talking about different things 🤔

Reply:
Oh, yes, i see that, will fix it.